From ab3dc0064726c42c86884accbf3ba08d52f6cca5 Mon Sep 17 00:00:00 2001
From: skodalix <srinivasx.kodali@intel.com>
Date: Mon, 29 Oct 2018 12:37:32 +0530
Subject: [PATCH] UGW_SW-32239: Fix for -Wsign-compare warning in userspace for
 including ethtool.h. It leads to compilation error when combining with
 -Werror

---
 include/uapi/linux/ethtool.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/uapi/linux/ethtool.h b/include/uapi/linux/ethtool.h
index e7e4e672d9a8..7ee86ebc1874 100644
--- a/include/uapi/linux/ethtool.h
+++ b/include/uapi/linux/ethtool.h
@@ -1488,7 +1488,7 @@ enum ethtool_link_mode_bit_indices {
 
 static inline int ethtool_validate_speed(__u32 speed)
 {
-	return speed <= INT_MAX || speed == SPEED_UNKNOWN;
+	return speed <= INT_MAX || (int32_t)speed == SPEED_UNKNOWN;
 }
 
 /* Duplex, half or full. */
